pkg/server/server.go

```diff
@@ -332,9 +339,14 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 
 // Close releases the server resources.
 func (s *Server) Close() {
-	if s.checkCache != nil {
-		s.checkCache.Stop()
+	if s.cachedCheckResolverCloser != nil {
+		s.cachedCheckResolverCloser()
 	}
+
+	if s.checkResolver != nil {
+		s.checkResolver.Close()
+	}
```

Comment:
```suggestion
	if s.checkQueryCacheEnabled {
		s.cachedCheckResolverCloser()
		s.checkResolver.Close()
	}
```

Reply:
The check `s.cachedCheckResolverCloser != nil` and `s.checkResolver != nil` are safer. Assuming a bug upstream, for example if we didn't construct the check resolver or check resolver closer properly when `checkQueryCacheEnabled=true`, then that wouldn't be a safer bet.

If the query cache is not enabled both _should be_ nil, but a direct check on nil is safer.